src/main/java/frc/team2412/robot/subsystem/ShooterVisionSubsystem.java

```diff
@@ -30,20 +45,47 @@ public boolean hasTarget() {
         return limelight.getEntry("tv").getDouble(0) == 1;
     }
 
-    // x-axis
+    /**
+     * Returns the yaw from the limelight to the hub.
+     *
+     * @return The yaw (horizontal rotation) in degrees.
+     */
     public double getYaw() {
         return limelight.getEntry("tx").getDouble(0);
     }
 
-    // returns in inches
+    /**
+     * Returns the distance from the limelight to the hub.
+     *
+     * @return THe distance in inches.
+     */
     public double getDistance() {
-        double distance = HEIGHT_TO_RIM / Math.tan(LIMELIGHT_ANGLE_OFFSET + getPitch());
-        return distance;
+        return HEIGHT_TO_RIM / Math.tan(Math.toRadians(LIMELIGHT_ANGLE_OFFSET + getPitch()));
     }
 
-    // y-axis
+    /**
+     * Returns the pitch from the robot to the hub.
+     *
+     * @return The pitch (vertical rotation) in degrees.
+     */
     public double getPitch() {
-        return limelight.getEntry("ty").getDouble(0);
+        return limelight.getEntry("ty").getDouble(DEFAULT_PITCH);
+    }
+
+    /**
+     * Returns the estimated robot pose as a {@link RigidTransform2}.
+     *
+     * The translation (inches) is relative to the hub, and the rotation has the same reference as
+     * STARTING_ROBOT_ROTATION. Positive rotation is clockwise, negative is counterclockwise.
+     *
+     * @return The estimated robot pose as a {@link RigidTransform2}.
+     */
+    public RigidTransform2 getEstimatedPose() {
+        double distanceToHub = getDistance();
+        Rotation2 robotAngle = gyro.getAdjustmentAngle().rotateBy(gyroAdjustmentAngle);
+        Rotation2 hubRobotAngle = robotAngle.rotateBy(Rotation2.fromDegrees(180 - getYaw()));
+        Vector2 translation = Vector2.fromAngle(hubRobotAngle).scale(distanceToHub);
+        return new RigidTransform2(translation, robotAngle);
```

Comment:
I'm not sure if `robotAngle` is accurate - from what I can see, it wouldn't update when the robot rotates, and it's documented as having the same reference as `STARTING_ROBOT_ROTATION`, meaning that it should change when the robot rotates.

Reply:
`getAdjustmentAngle()` was a typo, I meant `getUnadjustedAngle()`.